compiler/fir/raw-fir/fir-common/src/org/jetbrains/kotlin/fir/builder/ConversionUtils.kt

```diff
@@ -456,7 +457,12 @@ fun FirPropertyBuilder.generateAccessorsByDelegate(
 fun FirTypeRef.convertToArrayType(): FirUserTypeRef = buildUserTypeRef {
     source = this@convertToArrayType.source
     isMarkedNullable = false
-    qualifier += FirQualifierPartImpl(StandardClassIds.Array.shortClassName).apply {
+    val elementType = this@convertToArrayType as? FirUserTypeRef
+    val arrayQualifier =
+        if (elementType != null && elementType.qualifier.size == 1)
```

Comment:
I would say that this change might be not enough.
For example, it's not gonna work in case it has a form of `vararg x: kotlin.Int` or `vararg x: MyTypeAliasToInt`.
Maybe it's worth trying to move this logic to `org.jetbrains.kotlin.fir.resolve.transformers.FirTypeResolveTransformer#transformValueParameter` when the type is already resolved properly and we can just use it without guessing

BTW, something similar is already happening there at `result.transformVarargTypeToArrayType()`.
Looks like we have similar logic in two different places and I would say it's worth investigating which one should work for all cases.